include/model/CCategoryExamplesCollector.h

```diff
@@ -3,21 +3,22 @@
  * or more contributor license agreements. Licensed under the Elastic License;
  * you may not use this file except in compliance with the Elastic License.
  */
-#ifndef INCLUDED_ml_api_CCategoryExamplesCollector_h
-#define INCLUDED_ml_api_CCategoryExamplesCollector_h
+#ifndef INCLUDED_ml_model_CCategoryExamplesCollector_h
+#define INCLUDED_ml_model_CCategoryExamplesCollector_h
 
-#include <api/ImportExport.h>
-
-#include <core/CStatePersistInserter.h>
-#include <core/CStateRestoreTraverser.h>
+#include <model/ImportExport.h>
 
+#include <boost/container/flat_set.hpp>
```

Comment:
If you look at the "implementation decisions" section of the Doxygen comments it says why `flat_set` was used.  Then that got changed to a normal `set` because of some compiler incompatibility problem without updating the Doxygen comment.  Since I was moving the class I thought I'd update the comment and code to be in sync, and since our compilers can now all handle `flat_set` I thought I'd change the code rather than the comment.

For a single categorizer per job like we have at the moment it really doesn't matter - `flat_set` was a premature optimization.  However, in a future release we are considering adding per-partition categorization.  If this is done then the memory saving of `flat_set` could become noticeably beneficial.  (Say with 30 different types of logs in the Logs UI each with their own separate categorizer and hence separate examples.)